central/reports/service/v2/convert.go

```diff
@@ -18,3 +19,72 @@ func convertPrototoV2Reportstatus(status *storage.ReportStatus) *apiV2.ReportSta
 	}
 
 }
+
+func convertProtoReportCollectiontoV2(collection *storage.CollectionSnapshot) *apiV2.CollectionSnapshot {
+	if collection == nil {
+		return nil
+	}
+
+	return &apiV2.CollectionSnapshot{
+		Id:   collection.GetId(),
+		Name: collection.GetName(),
+	}
+}
+
+// ConvertProtoNotifierSnapshotToV2 converts notifiersnapshot proto to v2
+func ConvertProtoNotifierSnapshotToV2(notifierSnapshot *storage.NotifierSnapshot) *apiV2.NotifierSnapshot {
+	if notifierSnapshot == nil {
+		return nil
+	}
+	if notifierSnapshot.GetEmailConfig() == nil {
+		return &apiV2.NotifierSnapshot{}
+	}
+
+	emailConfig := &apiV2.EmailNotifierConfiguration{}
+	emailConfig.MailingLists = append(emailConfig.MailingLists, notifierSnapshot.GetEmailConfig().GetMailingLists()...)
+
+	notifierConfig := &apiV2.NotifierSnapshot_EmailConfig{
+		EmailConfig: emailConfig,
+	}
+
+	notifierName := notifierSnapshot.GetNotifierName()
+
+	return &apiV2.NotifierSnapshot{
+		NotifierName:   notifierName,
+		NotifierConfig: notifierConfig,
+	}
+}
+
+// convertPrototoV2ReportSnapshot converts storage.ReportSnapshot to apiV2.ReportSnapshot
+func convertProtoReportSnapshotstoV2(snapshots []*storage.ReportSnapshot) []*apiV2.ReportSnapshot {
+	if snapshots == nil {
+		return nil
+	}
+	res := make([]*apiV2.ReportSnapshot, 0, len(snapshots))
+	for _, snapshot := range snapshots {
+		snapshotv2 := &apiV2.ReportSnapshot{
+			ReportStatus:       convertPrototoV2Reportstatus(snapshot.GetReportStatus()),
+			Id:                 snapshot.GetReportId(),
+			Name:               snapshot.GetName(),
+			Description:        snapshot.GetDescription(),
+			CollectionSnapshot: convertProtoReportCollectiontoV2(snapshot.GetCollection()),
+			User: &apiV2.SlimUser{
+				Id:   snapshot.GetRequester().GetId(),
+				Name: snapshot.GetRequester().GetId(),
+			},
+			Schedule: reportConfig.ConvertProtoScheduleToV2(snapshot.GetSchedule()),
+			Filter: &apiV2.ReportSnapshot_VulnReportFilters{
+				VulnReportFilters: reportConfig.ConvertProtoVulnReportFiltersToV2(snapshot.GetVulnReportFilters()),
+			},
+		}
+		for _, notifier := range snapshot.GetNotifiers() {
+			converted := ConvertProtoNotifierSnapshotToV2(notifier)
+			snapshotv2.Notifiers = append(snapshotv2.Notifiers, converted)
```

Comment:
Since  `ConvertProtoNotifierSnapshotToV2` can return nil, only append to `snapshotv2.Notifiers` if `converted != nil`